s3/src/main/java/ch/cyberduck/core/s3/S3TokenExpiredResponseInterceptor.java

```diff
@@ -41,24 +40,24 @@ public class S3TokenExpiredResponseInterceptor extends DisabledServiceUnavailabl
     private static final int MAX_RETRIES = 1;
 
     private final S3Session session;
-    private final AWSProfileSTSCredentialsConfigurator configurator;
+    private final CredentialsConfigurator configurator;
 
-    public S3TokenExpiredResponseInterceptor(final S3Session session, final X509TrustManager trust, final X509KeyManager key, final LoginCallback prompt) {
+    public S3TokenExpiredResponseInterceptor(final S3Session session, final CredentialsConfigurator configurator) {
         this.session = session;
-        this.configurator = new AWSProfileSTSCredentialsConfigurator(trust, key, prompt);
+        this.configurator = configurator;
     }
 
     @Override
     public boolean retryRequest(final HttpResponse response, final int executionCount, final HttpContext context) {
         if(executionCount <= MAX_RETRIES) {
             switch(response.getStatusLine().getStatusCode()) {
                 case HttpStatus.SC_BAD_REQUEST:
-                    final S3ServiceException failure;
                     try {
                         if(null != response.getEntity()) {
                             EntityUtils.updateEntity(response, new BufferedHttpEntity(response.getEntity()));
-                            failure = new S3ServiceException(response.getStatusLine().getReasonPhrase(),
+                            final S3ServiceException failure = new S3ServiceException(response.getStatusLine().getReasonPhrase(),
                                     EntityUtils.toString(response.getEntity()));
+                            failure.setResponseCode(response.getStatusLine().getStatusCode());
```

Comment:
Why is this necessary now and was not necessary before?

Reply:
This was a bug in the existing code base filed as #10917.